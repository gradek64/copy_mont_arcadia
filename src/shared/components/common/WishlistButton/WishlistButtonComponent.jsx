import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import WishListIcon from '../WishList/WishListIcon'

class WishlistButtonComponent extends Component {
  static propTypes = {
    isSelected: PropTypes.bool,
    isAdding: PropTypes.bool,
    modifier: PropTypes.oneOf(['plp', 'pdp', 'quickview', 'minibag', 'bundle']),
    onClick: PropTypes.func,
    renderRemoveFromWishlistText: PropTypes.func,
    renderAddToWishlistText: PropTypes.func,
  }

  static defaultProps = {
    isAdding: false,
    isSelected: false,
  }

  renderWishlistText = () => {
    const {
      isSelected,
      renderAddToWishlistText,
      renderRemoveFromWishlistText,
    } = this.props

    if (isSelected) {
      return (
        renderRemoveFromWishlistText && (
          <span className="translate">{renderRemoveFromWishlistText()}</span>
        )
      )
    }
    return (
      renderAddToWishlistText && (
        <span className="translate">{renderAddToWishlistText()}</span>
      )
    )
  }

  render() {
    const { modifier, isSelected, onClick, isAdding } = this.props
    return (
      <button
        className={classNames(
          `WishlistButton`,
          'notranslate',
          modifier ? `WishlistButton--${modifier}` : null
        )}
        onClick={onClick}
      >
        <div>
          <span className="WishlistButton-icon">
            <WishListIcon
              id="wishlistIcon"
              modifier={modifier}
              isSelected={isSelected}
              isAdding={isAdding}
            />
          </span>
          {this.renderWishlistText()}
        </div>
      </button>
    )
  }
}

export default WishlistButtonComponent
