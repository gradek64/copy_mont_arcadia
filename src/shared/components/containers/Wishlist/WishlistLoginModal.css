/** @define WishlistLoginModal */

.WishlistLoginModal {
  &-forgetPassword {
    margin-bottom: 10px;
  }

  &-forms {
    display: flex;
    flex-direction: column;
  }

  &-header {
    margin-bottom: 20px;
  }

  &-heartIcon {
    display: block;
    background-image: $wishlist-header-icon;
    background-repeat: no-repeat;
    background-size: contain;
    background-position: center;
    margin: 10px auto;
    width: 40px;
    height: 36px;
  }

  &-loginForm,
  &-register {
    margin-bottom: 25px;
    margin-top: 0;
  }

  &-text {
    font-size: 18px;
    line-height: 1.22;
    text-align: center;
  }
}

@media $min-tablet {
  .WishlistLoginModal {
    &-formContainer {
      width: 50%;
      overflow:hidden;
    }

    &-forms {
      flex-direction: row;
      justify-content: center;
    }

    &-header {
      align-items: center;
      display: flex;
      flex-direction: row;
      justify-content: center;
    }

    &-heartIcon {
      margin: 0 20px;
      width: 23px;
      height: 21px;
    }
  }
}

@media $min-laptop {
  .WishlistLoginModal {
    &-formContainer {
      padding: 0 50px;
    }
  }
}
