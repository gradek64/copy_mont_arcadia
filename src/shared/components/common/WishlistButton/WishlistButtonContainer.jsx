import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import WishlistButtonComponent from './WishlistButtonComponent'

import WishlistLoginModal from '../../containers/Wishlist/WishlistLoginModal'

import {
  triggerWishlistLoginModal,
  addToWishlist,
  removeProductFromWishlist,
} from '../../../actions/common/wishlistActions'

class WishlistButtonContainer extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    isAddedToWishlist: PropTypes.bool.isRequired,
    modifier: PropTypes.oneOf(['plp', 'pdp', 'quickview', 'bundle', 'minibag'])
      .isRequired,
    productId: PropTypes.number.isRequired,
    addToWishlist: PropTypes.func.isRequired,
    triggerWishlistLoginModal: PropTypes.func.isRequired,
    removeProductFromWishlist: PropTypes.func.isRequired,
    onClickPreHook: PropTypes.func,
    afterAddToWishlist: PropTypes.func,
    onAuthenticationPreHook: PropTypes.func,
    onCancelLogin: PropTypes.func,
    renderRemoveFromWishlistText: PropTypes.func,
    renderAddToWishlistText: PropTypes.func,
    isFromQuickView: PropTypes.bool,
    productDetails: PropTypes.object.isRequired,
  }

  static defaultProps = {
    isAuthenticated: false,
    isAddedToWishlist: false,
    onClickPreHook: null,
    removeProductFromWishlistOverride: null,
    afterAddToWishlist: null,
    onAuthenticationPreHook: null,
    onCancelLogin: null,
    isFromQuickView: false,
  }

  // @NOTE isAddedToWishlist will be used only to initialize the 'selected' state
  constructor(props) {
    super(props)
    this.state = {
      isInWishlist: props.isAddedToWishlist,
      isAdding: false,
    }
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    // @NOTE Syncing local state `isInWishlist` with `isAddedToWishlist`
    if (
      this.props.isAddedToWishlist !== nextProps.isAddedToWishlist &&
      nextProps.isAddedToWishlist !== this.state.isInWishlist
    ) {
      this.setState({
        isInWishlist: nextProps.isAddedToWishlist,
      })
    }
  }

  handleWishlistClick = (e) => {
    e.preventDefault()
    const { isInWishlist } = this.state
    const {
      productId,
      isAuthenticated,
      triggerWishlistLoginModal,
      modifier,
      onAuthenticationPreHook,
      onClickPreHook,
      afterAddToWishlist,
      onCancelLogin,
      isFromQuickView,
      productDetails,
    } = this.props

    if (this.state.isAdding) return
    if (onClickPreHook) onClickPreHook()

    if (isAuthenticated) {
      return isInWishlist
        ? this.removeFromWishlistOptimistic()
        : this.addToWishlistOptimistic()
    }

    const actions = []
    if (onAuthenticationPreHook) actions.push(onAuthenticationPreHook())

    actions.push(
      triggerWishlistLoginModal(
        productId,
        WishlistLoginModal,
        modifier,
        afterAddToWishlist,
        onCancelLogin,
        isFromQuickView,
        productDetails
      )
    )

    return Promise.all(actions)
  }

  addToWishlistOptimistic = () => {
    const {
      productId,
      addToWishlist,
      modifier,
      afterAddToWishlist,
      productDetails,
    } = this.props

    this.setState({ isInWishlist: true, isAdding: true })

    return addToWishlist(productId, modifier, productDetails)
      .then(() => {
        if (afterAddToWishlist) return afterAddToWishlist()
      })
      .catch(() => {
        this.setState({ isInWishlist: false })
      })
      .then(() => {
        this.setState({ isAdding: false })
      })
  }

  removeFromWishlistOptimistic = () => {
    const {
      productId,
      modifier,
      removeProductFromWishlist,
      productDetails,
    } = this.props

    this.setState({ isInWishlist: false })

    return removeProductFromWishlist({
      productId,
      modifier,
      productDetails,
      reportToGA: true,
    }).catch(() => {
      this.setState({ isInWishlist: true })
    })
  }

  render() {
    const {
      modifier,
      renderAddToWishlistText,
      renderRemoveFromWishlistText,
    } = this.props
    return (
      <WishlistButtonComponent
        onClick={this.handleWishlistClick}
        isSelected={this.state.isInWishlist}
        isAdding={this.state.isAdding}
        modifier={modifier}
        renderAddToWishlistText={renderAddToWishlistText}
        renderRemoveFromWishlistText={renderRemoveFromWishlistText}
      />
    )
  }
}

const mapDispatchToProps = {
  triggerWishlistLoginModal,
  addToWishlist,
  removeProductFromWishlist,
}

export default connect(
  null,
  mapDispatchToProps
)(WishlistButtonContainer)
