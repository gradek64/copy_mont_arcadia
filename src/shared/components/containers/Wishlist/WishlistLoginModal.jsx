import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import Login from '../Login/Login'
import Register from '../Register/Register'
import ForgetPassword from '../ForgetPassword/ForgetPassword'

import {
  addToWishlistAfterLogin,
  captureWishlistEvent,
  getPaginatedWishlist,
} from '../../../actions/common/wishlistActions'
import {
  closeModal,
  clearModalChildren,
} from '../../../actions/common/modalActions'

import { getRoutePathWithParams } from '../../../selectors/routingSelectors'

if (process.browser) {
  require('./WishlistLoginModal.css')
}

@connect(
  (state) => ({
    relativePath: getRoutePathWithParams(state),
    isModalCancelled: state.modal.cancelled,
    pendingProductId: state.wishlist.pendingProductId,
  }),
  {
    addToWishlistAfterLogin,
    closeModal,
    getPaginatedWishlist,
    clearModalChildren,
    captureWishlistEvent,
  }
)
class WishlistLoginModal extends Component {
  static propTypes = {
    modifier: PropTypes.string,
    relativePath: PropTypes.string,
    pendingProductId: PropTypes.number,
    addToWishlistAfterLogin: PropTypes.func.isRequired,
    captureWishlistEvent: PropTypes.func,
    afterAddToWishlist: PropTypes.func,
    onCancelLogin: PropTypes.func,
    productDetails: PropTypes.object.isRequired,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  componentDidMount() {
    const { modifier, captureWishlistEvent } = this.props

    captureWishlistEvent('GA_WISHLIST_MODAL_PAGE_VIEW', { modifier })
  }

  componentWillUnmount() {
    const {
      modifier,
      isModalCancelled,
      captureWishlistEvent,
      onCancelLogin,
    } = this.props
    if (isModalCancelled) {
      captureWishlistEvent('GA_WISHLIST_MODAL_CLOSE', { modifier })
      if (onCancelLogin) onCancelLogin()
    }
  }

  handleSuccess = (authType) => {
    const { modifier, captureWishlistEvent } = this.props

    if (authType) {
      const action = `GA_WISHLIST_${authType.toUpperCase()}_SUCCESS`
      captureWishlistEvent(action)
    }

    switch (modifier) {
      case 'wishlist':
        return this.wishlistPageHandler()
      case 'plp':
      case 'pdp':
      case 'quickview':
      case 'minibag':
      case 'bundle':
        return this.addToWishlistHandler()
      default:
        return this.loginUserHandler()
    }
  }

  handleError = (authType) => {
    const { captureWishlistEvent } = this.props
    const action = `GA_WISHLIST_${authType.toUpperCase()}_ERROR`

    captureWishlistEvent(action)
  }

  loginUserHandler = () => {
    const { closeModal, clearModalChildren } = this.props
    const actions = [closeModal(), clearModalChildren()]
    return Promise.all([actions])
  }

  wishlistPageHandler = () => {
    const { closeModal, clearModalChildren, getPaginatedWishlist } = this.props
    const actions = [closeModal(), clearModalChildren(), getPaginatedWishlist()]
    return Promise.all([actions])
  }

  addToWishlistHandler = () => {
    const {
      addToWishlistAfterLogin,
      afterAddToWishlist,
      clearModalChildren,
      closeModal,
      modifier,
      pendingProductId,
      productDetails,
    } = this.props
    const actions = [clearModalChildren(), closeModal()]

    return addToWishlistAfterLogin(
      pendingProductId,
      modifier,
      productDetails
    ).then(() => {
      return Promise.all(actions).then(() => {
        if (afterAddToWishlist) afterAddToWishlist()
      })
    })
  }

  render() {
    const { l } = this.context
    const { relativePath } = this.props

    return (
      <div className="WishlistLoginModal">
        <div className="WishlistLoginModal-header">
          <i className="WishlistLoginModal-heartIcon" />
          <header className="WishlistLoginModal-text">{l`Sign in to save products to your Wishlist`}</header>
        </div>
        <div className="WishlistLoginModal-forms">
          <div className="WishlistLoginModal-formContainer">
            <Login
              className="WishlistLoginModal-loginForm"
              successCallback={() => this.handleSuccess('login')}
              errorCallback={() => this.handleError('login')}
              formName="wishlistLoginModal"
            />
            <ForgetPassword className="WishlistLoginModal-forgetPassword" />
          </div>
          <div className="WishlistLoginModal-formContainer">
            <Register
              className="WishlistLoginModal-register"
              source="WISHLIST"
              successCallback={() => this.handleSuccess('register')}
              errorCallback={() => this.handleError('register')}
              getNextRoute={() => relativePath}
              formName="wishlistLoginModal"
            />
          </div>
        </div>
      </div>
    )
  }
}

export default WishlistLoginModal
