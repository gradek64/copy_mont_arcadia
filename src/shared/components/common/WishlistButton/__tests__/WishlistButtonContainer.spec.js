import testComponentHelper from 'test/unit/helpers/test-component'

import WishlistButtonContainer from '../WishlistButtonContainer'
import WishlistLoginModal from '../../../containers/Wishlist/WishlistLoginModal'
import WishlistButtonComponent from '../WishlistButtonComponent'

describe('<WishlistButtonContainer />', () => {
  const productId = 123456
  const onClickPreHook = jest.fn()
  const modifier = 'pdp'
  const isFromQuickView = false
  const productDetails = {
    lineNumber: 'TX2345OL',
    price: '20.00',
  }
  const defaultProps = {
    isAddedToWishlist: false,
    isAuthenticated: false,
    productId,
    modifier,
    addToWishlist: jest.fn(() => Promise.resolve()),
    removeProductFromWishlist: jest.fn(() => Promise.resolve()),
    triggerWishlistLoginModal: jest.fn().mockReturnValue(Promise.resolve()),
    productDetails: {
      lineNumber: 'TX2345OL',
      price: '20.00',
    },
  }

  const event = { preventDefault: jest.fn() }

  const renderComponent = testComponentHelper(
    WishlistButtonContainer.WrappedComponent
  )

  beforeEach(() => {
    jest.clearAllMocks()
  })

  describe('@constructor', () => {
    it('sets initial state', () => {
      const { wrapper } = renderComponent(defaultProps)
      expect(wrapper.state('isInWishlist')).toBe(false)
    })
  })

  describe('@renders', () => {
    beforeEach(() => jest.clearAllMocks())

    it('should render default state', () => {
      const { getTree } = renderComponent({
        ...defaultProps,
      })
      expect(getTree()).toMatchSnapshot()
    })

    it('should render with plp modifier', () => {
      const { getTree } = renderComponent({
        ...defaultProps,
        modifier: 'plp',
      })
      expect(getTree()).toMatchSnapshot()
    })

    it('should pass down `isAdding` as false by default', () => {
      const { wrapper } = renderComponent(defaultProps)
      expect(wrapper.find(WishlistButtonComponent).prop('isAdding')).toBe(false)
    })
  })

  describe('@lifecycle', () => {
    describe('when item has been successfully added to wishlist', () => {
      beforeEach(() => jest.clearAllMocks())
      it('updates UI if does not reflect the value of `isAddedToWishlist` yet', () => {
        const { wrapper } = renderComponent({
          ...defaultProps,
          isAddedToWishlist: false,
          isAuthenticated: true,
        })
        expect(wrapper.state('isInWishlist')).toBe(false)
        wrapper.setProps({
          isAddedToWishlist: true,
        })
        expect(wrapper.state('isInWishlist')).toBe(true)
      })
    })
  })

  describe('@events', () => {
    describe('on clicking the wishlist button', () => {
      describe('button event', () => {
        it('should prevent default', async () => {
          const { instance } = renderComponent({
            ...defaultProps,
          })

          await instance.handleWishlistClick(event)
          expect(event.preventDefault).toHaveBeenCalledTimes(1)
        })
      })

      describe('when request is pending', () => {
        beforeEach(() => jest.clearAllMocks())

        it('should not trigger any action', async () => {
          const { instance } = renderComponent({
            ...defaultProps,
            isAddedToWishlist: false,
            isAuthenticated: true,
          })
          instance.setState({ isAdding: true })

          await instance.handleWishlistClick(event)
          expect(instance.props.addToWishlist).toHaveBeenCalledTimes(0)
          expect(
            instance.props.removeProductFromWishlist
          ).toHaveBeenCalledTimes(0)
          expect(
            instance.props.triggerWishlistLoginModal
          ).toHaveBeenCalledTimes(0)
        })

        it('should not call onClickPreHook', async () => {
          const { instance } = renderComponent({
            ...defaultProps,
            onClickPreHook,
            isAddedToWishlist: false,
            isAuthenticated: true,
          })
          instance.setState({ isAdding: true })

          await instance.handleWishlistClick(event)
          expect(instance.props.addToWishlist).toHaveBeenCalledTimes(0)
          expect(onClickPreHook).toHaveBeenCalledTimes(0)
          expect(
            instance.props.removeProductFromWishlist
          ).toHaveBeenCalledTimes(0)
          expect(
            instance.props.triggerWishlistLoginModal
          ).toHaveBeenCalledTimes(0)
        })
      })

      describe('when user is authenticated', () => {
        describe('and item is not in wishlist yet', () => {
          beforeEach(() => jest.clearAllMocks())

          it('should update the `isAdding` prop', () => {
            const { wrapper } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: false,
              isAuthenticated: true,
            })

            expect(wrapper.find(WishlistButtonComponent).prop('isAdding')).toBe(
              false
            )

            wrapper.find(WishlistButtonComponent).simulate('click', event)

            expect(wrapper.find(WishlistButtonComponent).prop('isAdding')).toBe(
              true
            )
          })

          it('should dispatch addToWishlist', async () => {
            const { instance, wrapper } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: false,
              isAuthenticated: true,
              modifier: 'plp',
            })

            expect(wrapper.state('isInWishlist')).toBe(false)
            await instance.handleWishlistClick(event)
            expect(wrapper.state('isInWishlist')).toBe(true)
            expect(instance.props.addToWishlist).toHaveBeenCalledTimes(1)
            expect(instance.props.addToWishlist).toHaveBeenCalledWith(
              productId,
              'plp',
              productDetails
            )
          })

          it('should call afterAddToWishlist', async () => {
            const { instance } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: false,
              isAuthenticated: true,
              modifier: 'plp',
              afterAddToWishlist: jest.fn().mockReturnValue(Promise.resolve()),
            })

            expect(instance.props.afterAddToWishlist).toHaveBeenCalledTimes(0)
            await instance.handleWishlistClick(event)
            expect(instance.props.afterAddToWishlist).toHaveBeenCalledTimes(1)
          })

          it('should revert local state if addToWishlist fails', async () => {
            const { instance, wrapper } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: false,
              isAuthenticated: true,
              addToWishlist: jest.fn(() => Promise.reject()),
            })

            expect(wrapper.state('isInWishlist')).toBe(false)
            await instance.handleWishlistClick(event)
            expect(wrapper.state('isInWishlist')).toBe(false)
          })
        })

        describe('and item has already been added to wishlist', () => {
          beforeEach(() => jest.clearAllMocks())

          it('should update the `isAdding` prop', () => {
            const { wrapper } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: true,
              isAuthenticated: true,
            })

            expect(wrapper.find(WishlistButtonComponent).prop('isAdding')).toBe(
              false
            )

            wrapper.find(WishlistButtonComponent).simulate('click', event)

            expect(wrapper.find(WishlistButtonComponent).prop('isAdding')).toBe(
              false
            )
          })

          it('should dispatch removeProductFromWishlist action', async () => {
            const { instance, wrapper } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: true,
              isAuthenticated: true,
              modifier: 'plp',
            })
            expect(wrapper.state('isInWishlist')).toBe(true)
            await instance.handleWishlistClick(event)
            expect(wrapper.state('isInWishlist')).toBe(false)
            expect(
              instance.props.removeProductFromWishlist
            ).toHaveBeenCalledTimes(1)
            expect(
              instance.props.removeProductFromWishlist
            ).toHaveBeenCalledWith({
              productId,
              modifier: 'plp',
              productDetails,
              reportToGA: true,
            })
          })

          it('should not call afterAddToWishlist', async () => {
            const { instance } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: true,
              isAuthenticated: true,
              modifier: 'plp',
              afterAddToWishlist: jest.fn().mockReturnValue(Promise.resolve()),
            })
            await instance.handleWishlistClick(event)
            expect(instance.props.afterAddToWishlist).toHaveBeenCalledTimes(0)
          })

          it('should revert local state if removeProductFromWishlist fails', async () => {
            const { instance, wrapper } = renderComponent({
              ...defaultProps,
              isAddedToWishlist: true,
              isAuthenticated: true,
              removeProductFromWishlist: jest.fn(() => Promise.reject()),
            })
            expect(wrapper.state('isInWishlist')).toBe(true)
            await instance.handleWishlistClick(event)
            expect(wrapper.state('isInWishlist')).toBe(true)
          })
        })
      })

      describe('when user is not authenticated', () => {
        beforeEach(() => jest.clearAllMocks())

        it('should dispatch triggerWishlistLoginModal action', async () => {
          const modifier = 'plp'
          const { productDetails } = defaultProps
          const { instance } = renderComponent({
            ...defaultProps,
            isAddedToWishlist: false,
            isAuthenticated: false,
            modifier,
          })
          await instance.handleWishlistClick(event)
          expect(
            instance.props.triggerWishlistLoginModal
          ).toHaveBeenCalledTimes(1)
          expect(instance.props.triggerWishlistLoginModal).toHaveBeenCalledWith(
            productId,
            WishlistLoginModal,
            modifier,
            null,
            null,
            isFromQuickView,
            productDetails
          )
        })

        it('should call onAuthenticationPreHook', async () => {
          const modifier = 'plp'
          const onAuthenticationPreHook = jest.fn()
          const { instance } = renderComponent({
            ...defaultProps,
            isAddedToWishlist: false,
            isAuthenticated: false,
            modifier,
            onAuthenticationPreHook,
          })
          expect(instance.props.onAuthenticationPreHook).toHaveBeenCalledTimes(
            0
          )
          await instance.handleWishlistClick(event)
          expect(instance.props.onAuthenticationPreHook).toHaveBeenCalledTimes(
            1
          )
        })

        it('passes afterAddToWishlist to modal', async () => {
          const modifier = 'plp'
          const { productDetails } = defaultProps
          const { instance } = renderComponent({
            ...defaultProps,
            isAddedToWishlist: false,
            isAuthenticated: false,
            modifier,
            afterAddToWishlist: jest.fn(),
          })
          await instance.handleWishlistClick(event)
          expect(
            instance.props.triggerWishlistLoginModal
          ).toHaveBeenCalledTimes(1)
          expect(instance.props.triggerWishlistLoginModal).toHaveBeenCalledWith(
            productId,
            WishlistLoginModal,
            modifier,
            instance.props.afterAddToWishlist,
            null,
            isFromQuickView,
            productDetails
          )
        })

        it('passes onCancelLogin to modal', async () => {
          const modifier = 'plp'
          const { productDetails } = defaultProps
          const { instance } = renderComponent({
            ...defaultProps,
            isAddedToWishlist: false,
            isAuthenticated: false,
            modifier,
            onCancelLogin: jest.fn(),
          })

          await instance.handleWishlistClick(event)
          expect(
            instance.props.triggerWishlistLoginModal
          ).toHaveBeenCalledTimes(1)
          expect(instance.props.triggerWishlistLoginModal).toHaveBeenCalledWith(
            productId,
            WishlistLoginModal,
            modifier,
            null,
            instance.props.onCancelLogin,
            isFromQuickView,
            productDetails
          )
        })
      })
    })
  })
})
