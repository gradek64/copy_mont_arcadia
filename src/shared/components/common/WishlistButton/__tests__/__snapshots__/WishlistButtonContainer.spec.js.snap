// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WishlistButtonContainer /> @renders should render default state 1`] = `
<WishlistButtonComponent
  isAdding={false}
  isSelected={false}
  modifier="pdp"
  onClick={[Function]}
/>
`;

exports[`<WishlistButtonContainer /> @renders should render with plp modifier 1`] = `
<WishlistButtonComponent
  isAdding={false}
  isSelected={false}
  modifier="plp"
  onClick={[Function]}
/>
`;
